### PR TITLE
Pure scalar tensor fusion

### PR DESCRIPTION
**Pull request by jjsjann123**
Tiny fix to allow fusion with pure scalar tensor

### CODE REVIEW

File: test/cpp/jit/test_gpu.cpp [hunk 1]
```diff
@@ -14013,6 +14013,37 @@ TEST(NVFuserTest, FusionPredicatedBlockBroadcast_CUDA) {
...
+
+  at::Tensor cg_output = at::empty({}, options);
+
+  scheduleFusion(&fusion, {input});
+
+  FusionExecutor fe;
+  fe.compileFusion(&fusion);
+  fe.runFusion({input}, {cg_output});
+  fe.runFusion({input}, {cg_output});
+  fe.runFusion({input}, {cg_output});
```
Comment (csarofeen) [inline test/cpp/jit/test_gpu.cpp]: Why run 3 times?

File: torch/csrc/jit/codegen/cuda/scheduler/pointwise.cpp [hunk 2]
```diff
@@ -39,8 +39,10 @@ bool scheduleFusion(Fusion* fusion) {
   // Run through outputs, grab all inputs of outputs
   // squeeze with computeAt to set overall structure.
   for (auto output : fusion->outputs()) {
-    if (output->getValType() != ValType::TensorView)
+    if (output->getValType() != ValType::TensorView ||
+        output->as<TensorView>()->nDims() == 0) {
```
Comment (csarofeen) [inline torch/csrc/jit/codegen/cuda/scheduler/pointwise.cpp:43]: I likely need to consider this kind of logic in reduction/normalization schedulers.

**Review by csarofeen — APPROVED** [review]
This PR looks good, though I'm worried our schedulers aren't generally prepared for 0-dim tensors. Need to consider.

File: torch/csrc/jit/codegen/cuda/scheduler/pointwise.cpp (hunk 2 above)
Comment (jjsjann123) [reply torch/csrc/jit/codegen/cuda/scheduler/pointwise.cpp:43]: Yes, AFAIK, I haven't enabled reduce to a scalar tensor yet.
Feel free to leave it as-is and I can patch them when I get to that.

File: test/cpp/jit/test_gpu.cpp (hunk 1 above)
Comment (jjsjann123) [reply test/cpp/jit/test_gpu.cpp]: Oops, I was doing too many python tests and copied execution out of bad habits :stuck_out_tongue:

**Comment by jjsjann123** [comment]
Tests passed, will merge after CI clear

**Pull request merged**